### DISCUSSION TITLE
API Method For Clearing a Feature From Storage

### DISCUSSION THREAD

**Discussion by mikejr83** [Michsior14/ngrx-addons#723] [Ideas]
I have an expensive query that I perform on user log-in that I am storing in state. I'm storing the entity collection in state. Everything is great. 

Something that I'm trying to figure out, however, is how to clear the persisted state for that collection. I'd like to clear this on user log out. I tried using an effect on the logout action to trigger a clear action on the state. In the reducer the state is set back to its initial state. However, it appears that this is not getting persisted to localstorage. I'm not sure if its a timing issue since I'm invoking a call to my auth provider's library when a subscription fires on a combine latest of both the logout and clear actions fire.

I could go directly to local storage and kill the key, but I feel like a service or some other API supported way would be more beneficial. Obviously, if there is a better way to handle this situation I'd be open to changing things up.

Thanks!

**Comment by Michsior14** [comment]
Hey!

The situation you are describing should be actually already handled, since once you set the state then the persist state reducer kicks in (as long as you don't have skip/filters applied). I would assume it's the timing issue? Can you try to [debug it via devtools](https://ngrx.io/api/store-devtools/StoreDevtoolsConfig#logOnly), e.g. `logOnly` and `trace`? (This make me wonder if debug option should be also provided for persist/sync state).

Regarding the storage access I am not entirely sure how to tackle this as each state (either global or per feature) can have totally different implementation. We could create the service that will require the state key to operate on the storage although I wonder how really helpful it will be as this could lead to malformed state in some cases.
